Add override list to admin Information page

unlocomqx opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add override list to the information page
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #21368 
| How to test?  | Open the backoffice Information page, it should be displayed like so ![image](https://user-images.githubusercontent.com/793712/95510499-ede2fb80-09ad-11eb-938f-8f34e7893c5e.png)

I made it the easy way, maybe there is a better way to fetch effective overrides and not just the php files list
If you know, please let me know

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/21375)
<!-- Reviewable:end -->

unlocomqx committed:
Add override list to admin Information page

PierreRambaud commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig:
```diff
@@ -102,6 +102,21 @@
+              </ul>
+            </div>
+          </div>
+        </div>
     </div>
```
```suggestion
        </div>
      </div>
```

unlocomqx commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig:
```diff
@@ -117,7 +117,7 @@
           </div>
         </div>
-    </div>
+      </div>
```
I just noticed that this div closing tag should be outside the %if statement, can you confirm?

unlocomqx replied:
![image](https://user-images.githubusercontent.com/793712/95565672-71393700-0a18-11eb-98e2-7b579c952082.png)

PierreRambaud replied:
Maybe it's time to change the whole indentation of this file and use 2spaces? :D

unlocomqx replied:
Will gladly do it, I will rebase and push

unlocomqx committed:
Close column div correctly

unlocomqx reviewed (commented):
The closing div should be moved outside the %if in order to close the col-lg-6 correctly

PierreRambaud commented on src/Adapter/Shop/ShopInformation.php:
```diff
@@ -27,6 +27,7 @@
 namespace PrestaShop\PrestaShop\Adapter\Shop;
 
 use AppKernel;
+use Tools;
 use PrestaShop\PrestaShop\Adapter\LegacyContext;
```
```suggestion
use AppKernel;
use PrestaShop\PrestaShop\Adapter\LegacyContext;
use Tools;
```

NeOMakinG commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig:
```diff
@@ -27,261 +27,275 @@
+    <div class="card" id="checkConfiguration">
+      <h3 class="card-header">
+        <i class="material-icons">info_outline</i> {{ 'Check your configuration'|trans }}
+      </h3>
+      <div class="card-block">
+        <div class="card-text">
+          <p>
+            <strong>{{ 'Required parameters:'|trans }}</strong>
+            {% if requirements.failRequired == false %}
+            <span class="text-success">{{ 'OK'|trans({}, 'Admin.Advparameters.Notification') }}</span>
+          </p>
+          {% else %}
+          <span class="text-danger">{{ 'Please fix the following error(s)'|trans({}, 'Admin.Advparameters.Notification') }}</span>
+          </p>
+          <ul>
+            {% for key, value in requirements.testsRequired %}
+            {% if 'fail' == value %}
+            <li>{{ requirements.testsErrors[key] }}</li>
+            {% endif %}
+            {% endfor %}
+          </ul>
+          {% endif %}
+          {% if requirements.failOptional is defined %}
+          <p>
+            <strong>{{ 'Optional parameters:'|trans }}</strong>
+            {% if requirements.failOptional == false %}
+            <span class="text-success">{{ 'OK'|trans({}, 'Admin.Advparameters.Notification') }}</span>
+          </p>
+          {% else %}
+          <span class="text-danger">{{ 'Please fix the following error(s)'|trans({}, 'Admin.Advparameters.Notification') }}</span>
+          </p>
+          <ul>
+            {% for key, value in requirements.testsOptional %}
+            {% if 'fail' == value %}
+            <li>{{ requirements.testsErrors[key] }}</li>
+            {% endif %}
+            {% endfor %}
+          </ul>
+          {% endif %}
+          {% endif %}
```
This is so unreadable :(

unlocomqx replied:
The IDE looks to be having a hard time formatting the file, I will try to do it manually

unlocomqx replied:
I found an unclosed p tag in the mail info, I re-wrote it and simplified it a bit

NeOMakinG replied:
Already way better :)

unlocomqx committed:
Reorder imports

unlocomqx committed:
Reindent with 2 spaces

unlocomqx committed:
Fix mail info section

Progi1984 commented on src/Adapter/Shop/ShopInformation.php:
```diff
@@ -61,4 +62,14 @@ public function getShopInformation()
+    /**
+     * @return array
+     */
+    public function getOverridesList()
```
```suggestion
    public function getOverridesList(): array
```

unlocomqx replied:
Added in https://github.com/PrestaShop/PrestaShop/pull/21375/commits/c3bc31e04cc44b18533f624938134b4a581f8011

Progi1984 commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig:
```diff
@@ -27,261 +27,277 @@
+        {% endfor %}
+        </ul>
+      </div>
+      </div>
```
Please indent code :)

unlocomqx replied:
Done in https://github.com/PrestaShop/PrestaShop/pull/21375/commits/c5bcc4f5c14646bedb1fb0ac8ae7e53fc689a635

Progi1984 reviewed (commented):
A small comment but it's a good job.

unlocomqx committed:
Add type annotation to getOverridesList method

unlocomqx committed:
Indent overrides block

unlocomqx committed:
Simplify configuration check block

Cutting html blocks with an if statement is unreadable and makes IDEs struggle to format/indent the code. It is simpler to just duplicate the code.

unlocomqx commented:
Sorry, I had to simplify the configuration check section. The if statement was "cutting" a p tag which made the code unreadable!

LouiseBonnard commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig:
```diff
@@ -27,261 +27,281 @@
+      </h3>
+      <div class="card-block">
+        <div class="card-text">
+          <p>{{ 'This information must be provided when you report an issue on our bug tracker or forum.'|trans }}</p>
```
How about updating this part and mentioning GitHub instead of 'our bug tracker'?

Progi1984 replied:
If we change our bug tracker, no need to change the wording ;)

LouiseBonnard replied:
Well, the 'bug tracker' used to refer to the Forge so it might be a little confusing for the users, especially now that the footer has been removed from the back office.

unlocomqx replied:
Whatever you decide guys. You both made good points!

Progi1984 replied:
You're right @LouiseBonnard. Ok for me for Github ;)

unlocomqx replied:
Resolved in af9eade202539b38dd272839ce34570b813373d0

LouiseBonnard commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig:
```diff
@@ -27,261 +27,281 @@
+          <div class="card-text">
+            {% if system.uname is not empty %}
+              <p>
+                <strong>{{ 'Server information'|trans }}</strong> {{ system.uname }}
```
```suggestion
                <strong>{{ 'Server information:'|trans }}</strong> {{ system.uname }}
```

LouiseBonnard replied:
For consistency, I suggest to add `:` at the end. See the screenshot below:

<img width="759" alt="Capture d’écran 2020-10-21 à 11 40 07" src="https://user-images.githubusercontent.com/32565890/96702930-8d32c600-1392-11eb-8035-6dc4d9868e80.png">

unlocomqx replied:
Resolved in 9ca32e72

LouiseBonnard commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig:
```diff
@@ -27,261 +27,281 @@
+    $(document).ready(function()
+    {
+      var translations = {
+        missing: '{{ "Missing files"|trans({}, "Admin.Advparameters.Notification")|e('js') }}',
+        updated: '{{ "Updated files"|trans({}, "Admin.Advparameters.Notification")|e('js') }}',
+        changesDetected: '{{ "Changed/missing files have been detected."|trans({}, "Admin.Advparameters.Notification")|e('js') }}',
+        noChangeDetected: '{{ "No change has been detected in your files."|trans({}, "Admin.Advparameters.Notification")|e('js') }}'
```
Watch the commas (`"`), they look different from the usual `'`.

unlocomqx replied:
I fixed the quotes in 7e609646b007eba4e2e1a87926c3305f85cbd744

unlocomqx committed:
Add colon to info header for consistency with other headers

unlocomqx committed:
Replace double quotes with single quotes in twig translations

unlocomqx committed:
Add brackets to if blocks

unlocomqx committed:
Change information section wording

atomiix commented on src/PrestaShopBundle/Resources/views/Admin/Configure/AdvancedParameters/system_information.html.twig:
```diff
@@ -27,261 +27,282 @@
+      <h3 class="card-header">
+        <i class="material-icons">info_outline</i> {{ 'List of overrides'|trans({}, 'Admin.Advparameters.Feature') }}
+      </h3>
+      <div class="card-block">
+      <div class="card-text">
+        <ul>
+          {% for override in system.overrides %}
+            <li>{{ override }}</li>
+          {% endfor %}
+        </ul>
+      </div>
+      </div>
```
```suggestion
      <div class="card-block">
        <div class="card-text">
          <ul>
            {% for override in system.overrides %}
              <li>{{ override }}</li>
            {% endfor %}
          </ul>
        </div>
      </div>
```

unlocomqx committed:
Fix template indentation

Co-authored-by: atomiix <tbaccelli@gmail.com>

khouloudbelguith commented:
Hi @unlocomqx,

It is ok :heavy_check_mark: 
![image](https://user-images.githubusercontent.com/16067358/98373863-26730500-2040-11eb-880d-cd6ebf677dd1.png)
![image](https://user-images.githubusercontent.com/16067358/98373891-2ffc6d00-2040-11eb-8bba-76cf52b47b09.png)

https://drive.google.com/file/d/1DOsEu4b1nqAOvDrJLd4a6GE9I0RpA4BU/view?usp=sharing

Thanks!

matks commented:
Thank you @unlocomqx
